test/test-lib/fixtures.ts

```diff
@@ -113,6 +148,88 @@ const loaders: Dictionary<LoaderFunc> = {
 	},
 };
 
+loaders['devices'] = async (jsonData, fixtures) => {
+	const user = await fixtures.users[jsonData.belongs_to__user];
+	if (user == null) {
+		logErrorAndThrow(`Could not find user: ${jsonData.user}`);
+	}
+	const application = await fixtures.applications[
+		jsonData.belongs_to__application
+	];
+	if (application == null) {
+		logErrorAndThrow(
+			`Could not find application: ${jsonData.belongs_to__application}`,
+		);
+	}
+
+	const requestOpts = buildApiRequestOpts({
+		path: 'resin/device',
+		body: {
+			belongs_to__application: application.id,
+			belongs_to__user: user.id,
+			..._.pick(
+				jsonData,
+				'uuid',
+				'device_type',
+				'is_managed_by__device',
+				'os_version',
+				'supervisor_version',
+				'logs_channel',
+				'custom_latitude',
+				'custom_longitude',
+				'os_variant',
+				'os_version',
+				'supervisor_version',
+			),
+		},
+		user,
+	});
+
+	requestOpts.body.is_of__device_type = (
+		await fixtures.deviceTypes[jsonData.device_type]
+	).id;
+	delete requestOpts.body.device_type;
+
+	// keeping this for now, because we will need it after the migration PR
+	// requestOpts.body.device_type_slug = requestOpts.body.device_type;
+	// delete requestOpts.body.device_type;
+
+	const [response, body] = await requestAsync(requestOpts);
+
+	if (response.statusCode !== 201) {
+		logErrorAndThrow('Failed to create device', jsonData, body);
+	}
+
+	// The is_online field is not in the create whitelist,
+	// so we need to do a separate patch after the device gets created
+	if (jsonData.is_online) {
+		await api.resin.patch({
+			resource: 'device',
+			passthrough: { req: root },
+			id: body.id,
+			body: { is_online: true },
+		});
+
+		body.is_online = true;
+	}
```

Comment:
There's no such thing in open-balena-api so you can bundle it in the single request above